src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy

```diff
@@ -1369,13 +1407,786 @@ class CookieSyncSpec extends BaseSpec {
+        assert genericBidderStatus?.userSync?.type == REDIRECT
+        assert HttpUtil.findUrlParameterValue(genericBidderStatus.userSync?.url, "f") == PIXEL.name
+
+        and: "Response should contain '#formatParam' format parameter"
+        def rubiconBidderStatus = response.getBidderUserSync(RUBICON)
+        assert rubiconBidderStatus?.userSync?.type == IFRAME
+        assert HttpUtil.findUrlParameterValue(rubiconBidderStatus.userSync?.url, "f") == BLANK.name
+
+        and: "Response should contain coop-synced bidder"
+        assert response.bidderStatus.bidder.containsAll(ADKERNEL, ACUITYADS, ACEEX, APPNEXUS, AAX, RUBICON, OPENX, GENERIC)
+    }
+
+    def "PBS cookie sync request should filter bidder due to filterSettings"() {
+        given: "Empty cookie sync request body"
+        def filterSettings = new FilterSettings().tap {
+            iframe = new MethodFilter().tap {
+                filter = INCLUDE
+                bidders = [RUBICON, GENERIC]
+            }
+            image = new MethodFilter().tap {
+                filter = EXCLUDE
+                bidders = [APPNEXUS, OPENX]
+            }
+        }
+        def cookieSyncRequest = new CookieSyncRequest(filterSettings: filterSettings, limit: 0)
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Bidder should be exclude by filter"
+        def rubiconUserSync = response.getBidderUserSync(RUBICON)
+        assert rubiconUserSync?.userSync?.type == IFRAME
+
+        and: "Bidder shouldn't be include by filter"
+        assert !response.getBidderUserSync(APPNEXUS)
+    }
+
+    def "PBS cookie sync request should include all bidder due to filterSettings"() {
+        given: "Empty cookie sync request body"
+        def filterSettings = new FilterSettings().tap {
+            iframe = new MethodFilter(bidders: [RUBICON], filter: INCLUDE)
+            image = new MethodFilter(bidders: [APPNEXUS], filter: INCLUDE)
+        }
+        def cookieSyncRequest = new CookieSyncRequest(filterSettings: filterSettings, limit: 0)
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Bidder should be include by filter"
+        assert response.getBidderUserSync(APPNEXUS)
+        assert response.getBidderUserSync(RUBICON)
+    }
+
+    def "PBS cookie sync request should exclude all iframe bidders when asterisk present in bidders filterSettings"() {
+        given: "Empty cookie sync request body"
+        def filterSettings = new FilterSettings().tap {
+            iframe = new MethodFilter(bidders: "*", filter: EXCLUDE)
```

Comment:
Can we make this `"*"` as a constant? So that it is clear from the outside what it is responsible for. The more it repeats itself.